monai/apps/reconstruction/networks/nets/complex_unet.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Union
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+
+from monai.apps.reconstruction.networks.nets.utils import (
+    complex_normalize,
+    reshape_channel_complex_to_last_dim,
+    reshape_complex_to_channel_dim,
+)
+from monai.networks.nets.basic_unet import BasicUNet
+from monai.transforms import DivisiblePad
+
+
+class ComplexUnet(nn.Module):
+    """
+    This variant of U-Net handles complex-value input/output. It can be
+    used as a model to learn sensitivity maps in multi-coil MRI data. It is
+    built based on :py:class:`monai.networks.nets.BasicUNet`.
+    It also applies default normalization to the input which makes it more stable to train.
+
+    The data being a (complex) 2-channel tensor is a requirement for using this model.
+
+    Modified and adopted from: https://github.com/facebookresearch/fastMRI
+
+    Args:
+        spatial_dims: number of spatial dimensions.
+        features: six integers as numbers of features. denotes number of channels in each layer.
+        act: activation type and arguments. Defaults to LeakyReLU.
+        norm: feature normalization type and arguments. Defaults to instance norm.
+        bias: whether to have a bias term in convolution blocks. Defaults to True.
+        dropout: dropout ratio. Defaults to 0.0.
+        upsample: upsampling mode, available options are
+            ``"deconv"``, ``"pixelshuffle"``, ``"nontrainable"``.
+        pad_factor: an integer denoting the number which each padded dimension will be divisible to.
+            For example, 16 means each dimension will be divisible by 16 after padding
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int = 2,
+        features: Sequence[int] = (32, 32, 64, 128, 256, 32),
+        act: Union[str, tuple] = ("LeakyReLU", {"negative_slope": 0.1, "inplace": True}),
+        norm: Union[str, tuple] = ("instance", {"affine": True}),
+        bias: bool = True,
+        dropout: Union[float, tuple] = 0.0,
+        upsample: str = "deconv",
+        pad_factor: int = 16,
+    ):
+        super().__init__()
+        self.unet = BasicUNet(
```

Comment:
looks like this could be any fully convolutional network as long as the input and output channels are 2. if that's true, would be nice to make it a configurable part:

```py
class ComplexUnet(nn.Module):
    def __init__(
        ...,
        conv_net=None,
        ...
    ):
        if conv_net is None:
            self.unet = BasicUNet(...)
        else:
            self.unet = conv_net
        ...
```